### DISCUSSION TITLE
Help how can I write an array of this type

### DISCUSSION THREAD

**Discussion by hpvcodeDev** [ForNeVeR/wpf-math#476] [Q&A]
Hello, I would like to know if you can write these types of matrices, that is, place a vertical and horizontal line like in Latex, I have tried every possible way but I did not achieve my purpose. I am working in c# wpf WpfMath

`M =\left[
\begin{array}{c c c |c}
1 & 2 & 2 & 2\\ 
3 & 4 & 2 & 2\\ \hline
5 & 6 & 2 & 2\\ 
\end{array} \right] `

`M =\left[
\begin{array}{c c c :c}
1 & 2 & 2 & 2\\ 
3 & 4 & 2 & 2\\ \hdashline
5 & 6 & 2 & 2\\ 
\end{array} \right] `

![lagrida_latex_editor](https://github.com/ForNeVeR/xaml-math/assets/129473476/472ff87e-97a3-481c-9a83-f1165ff1dec0)
![image](https://github.com/ForNeVeR/xaml-math/assets/129473476/5ff4ecb9-9f6f-4567-bbc4-553ccef34627)

**Comment by ForNeVeR** [comment]
I don't think this is currently supported, and I've no idea how to emulate that in the currently supported subset of markup. Created a feature request #477.

**Reply by hpvcodeDev** [reply]
It would be great if you could incorporate it, since in latex it is used for example to generate tables, I also don't know if you can add indexes to the headers of a matrix, for example in the image I achieved that effect using wpfmath combination of matrices but the upper part It doesn't align with my matrix in each column. I don't know if that can be done or if you could implement it.

`\matrix{{\matrix{y_1 & \theta_1 & y_2 & \theta_2 }} &  {} \\  {\begin{pmatrix} a_1 & a_2 & a_3  & a_4\\ b_1 & b_2 & b_3 & b_4 \\ c_1 & c_2 & c_3 & c_4 \\ d_1 & d_2 & d_3 & d_4 \end{pmatrix}} & {\matrix{y_1 \\ \theta_1 \\ y_2 \\ \theta_2 }} }`

![image](https://github.com/ForNeVeR/xaml-math/assets/129473476/66365988-ae8d-45c5-94ff-07fc78616de7)
